jax/_src/lax/lax.py

```diff
@@ -81,13 +81,19 @@ def _try_broadcast_shapes(shapes):
   return tuple(result_shape)
 
 @cache()
+def _broadcast_shapes(*shapes):
+  ndim = _max(len(shape) for shape in shapes)
+  shapes = [(1,) * (ndim - len(shape)) + shape for shape in shapes]
+  return _try_broadcast_shapes(shapes)
+
 def broadcast_shapes(*shapes):
   """Returns the shape that results from NumPy broadcasting of `shapes`."""
+  if not shapes:
+    return ()
   if len(shapes) == 1:
     return shapes[0]
-  ndim = _max(len(shape) for shape in shapes)
-  shapes = [(1,) * (ndim - len(shape)) + shape for shape in shapes]
-  result_shape = _try_broadcast_shapes(shapes)
+  shapes = [(shape,) if np.ndim(shape) == 0 else tuple(shape) for shape in shapes]
```

Comment:
We need to normalize shapes to tuples, in order to support integers as well as lists and 1D numpy arrays which would otherwise break the caching decorator since they are not hashable.